frontend/src/metabase/components/AppBanner/AppBanner.tsx

```diff
@@ -12,12 +9,21 @@ export const AppBanner = () => {
   const tokenStatus = useSetting("token-status");
   const readOnly = useSetting("read-only-mode");
 
-  if (tokenStatus && shouldRenderPaymentBanner({ isAdmin, tokenStatus })) {
-    return <PaymentBanner isAdmin={isAdmin} tokenStatus={tokenStatus} />;
+  const paymentStatuses = ["past-due", "unpaid", "invalid"];
+  const shouldRenderPaymentBanner =
+    tokenStatus && paymentStatuses.includes(tokenStatus?.status ?? "");
+
+  if (!isAdmin) {
```

Comment:
Even if it's correct. I'm still confused with 2 returns with `null`. What's the last `return null` in this version supposed to mean? I found the previous version easier to reason about because there're 2 valid conditions then the rest. Now I feel the conditions are a bit all over the place.

Reply:
sorry that I'm late to this party, but I think the read only banner should be visible to all users in an instance. I think it's a bug that the backend doesn't expose that setting to the frontend.